### DISCUSSION TITLE
Cannot get workflow diagnostics

### DISCUSSION THREAD

**Discussion by vshalt** [LazyVim/LazyVim#2291] [Q&A]
I have not added any config to the lazyvim configuration regarding any LSP servers. I have installed the servers with Mason.

I want to get project wide diagnostics, currently I get diagnostics of the current buffer, if the buffer is closed the diagnostics disappear.
Is there a way to turn on diagnostics for the whole project.
I usually change to the root of the project and open lazyvim with `nvim .` It is only when I open a buffer does the lsp server start (which is expected).

There is an option in Whichkey which toggles project wide diagnostics. How can I get the diagnostics to work. Any snippets to make this work are  appreciated. TIA

**Comment by dpetka2001** [comment]
I believe that is entirely dependent on the LSP server itself. For example, I'm pretty sure `tsserver` supports workspace diagnostics only for files that are loaded into buffers (and I think the same holds true for `lua_ls`). You should honestly read the docs about the LSP server you want to use, to see if it supports "true" workspace diagnostics or not.
